erizo/src/erizo/stats/StatNode.h

```diff
@@ -0,0 +1,115 @@
+#ifndef ERIZO_SRC_ERIZO_STATS_STATNODE_H_
+#define ERIZO_SRC_ERIZO_STATS_STATNODE_H_
+
+#include <string>
+#include <map>
+#include <memory>
+
+#include "lib/Clock.h"
+
+namespace erizo {
+
+class StatNode {
+ public:
+  StatNode() {}
+  virtual ~StatNode() {}
+
+  virtual StatNode& operator[](std::string key);
+
+  virtual StatNode& operator[](uint64_t key) { return (*this)[std::to_string(key)]; }
+
+  template <typename Node>
+  void insertStat(std::string key, Node&& stat) {  // NOLINT
+    node_map_.insert(std::make_pair(key, std::make_shared<Node>(std::forward<Node>(stat))));
```

Comment:
Can you explain the use of std::forward here? Maybe even leave a comment so we don't break this in the future